controllers/loadtest_controller.go

```diff
@@ -175,131 +176,221 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
+					logWithServer.Error(updateErr, "failed to update status after failure to create pod for server")
+				}
+				return *result, err
+			}
 		}
+		for i := range missingPods.Clients {
+			logWithClient := log.WithValues("client", missingPods.Clients[i])
+
+			pod, err := builder.PodForClient(&missingPods.Clients[i])
+			if err != nil {
+				logWithClient.Error(err, "failed to construct a pod struct for supplied client struct")
+				test.Status.State = grpcv1.Errored
+				test.Status.Reason = grpcv1.ConfigurationError
+				test.Status.Message = fmt.Sprintf("failed to construct a pod for client at index %d: %v", i, err)
+				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
+					logWithClient.Error(updateErr, "failed to update status after failure to construct a pod for client")
+				}
+				return ctrl.Result{}, err
+			}
 
-		result, err := createPod(pod)
-		if result != nil && !kerrors.IsAlreadyExists(err) {
-			logWithDriver.Error(err, "failed to create pod for driver")
+			if missingPods.Clients[i].Pool == nil {
+				pod.Labels[config.PoolLabel] = defaultClientPool
+			} else {
+				pod.Labels[config.PoolLabel] = *missingPods.Clients[i].Pool
+			}
 
-			test.Status.State = grpcv1.Errored
-			test.Status.Reason = grpcv1.KubernetesError
-			test.Status.Message = fmt.Sprintf("failed to create pod for driver: %v", err)
+			result, err := createPod(pod)
+			if result != nil && !kerrors.IsAlreadyExists(err) {
+				logWithClient.Error(err, "failed to create pod for client")
+				test.Status.State = grpcv1.Errored
+				test.Status.Reason = grpcv1.KubernetesError
+				test.Status.Message = fmt.Sprintf("failed to create pod for client at index %d: %v", i, err)
+				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
+					logWithClient.Error(updateErr, "failed to update status after failure to create pod for client")
+				}
+				return *result, err
+			}
+		}
+		if missingPods.Driver != nil {
+			logWithDriver := log.WithValues("driver", missingPods.Driver)
+
+			pod, err := builder.PodForDriver(missingPods.Driver)
+			if err != nil {
+				logWithDriver.Error(err, "failed to construct a pod struct for supplied driver struct")
+				test.Status.State = grpcv1.Errored
+				test.Status.Reason = grpcv1.ConfigurationError
+				test.Status.Message = fmt.Sprintf("failed to construct a pod for driver: %v", err)
+				if updateErr := r.Status().Update(ctx, test); updateErr != nil {
+					logWithDriver.Error(updateErr, "failed to update status after failure to construct a pod for driver")
+				}
+				return ctrl.Result{}, err
+			}
 
-			if updateErr := r.Status().Update(ctx, test); updateErr != nil {
-				logWithDriver.Error(updateErr, "failed to update status after failure to create pod for driver")
+			if missingPods.Driver.Pool == nil {
+				pod.Labels[config.PoolLabel] = defaultDriverPool
```

Comment:
It may be a nice enhancement to pass the default pools we calculate into the podbuilder and allow it to set those.